amundsen_application/static/js/components/SearchPage/SearchFilter/index.tsx

```diff
@@ -1,168 +1,153 @@
-            checked: false, // pull value or infer value from state
-            count: 100, // pull value from state
-          }
-        ]
-      },
-      {
-        title: 'Badges', // category.displayName
-        categoryId: 'badges', // category.id
-        inputProperties: [
-          {
-            value: 'sla', // value.id
-            labelText: 'Missed SLA', // value.displayName
-            checked: true, // pull value or infer value from state
-            count: 3, // pull value from state
-          },
-          {
-            value: 'quality', // value.id
-            labelText: 'High Quality', // value.displayName
-            checked: true, // pull value or infer value from state
-            count: 12, // pull value from state
-          },
-          {
-            value: 'pii', // value.id
-            labelText: 'PII', // value.displayName
-            checked: false, // pull value or infer value from state
-            count: 34, // pull value from state
-          },
-          {
-            value: 'deprecated', // value.id
-            labelText: 'Deprecated', // value.displayName
-            checked: false, // pull value or infer value from state
-            count: 3, // pull value from state
-          }
-        ]
+  const resourceType = state.search.selectedTab;
+  const filterCategories = getFilterConfigByResource(resourceType);
+  const filterState = state.search.filters;
+
+  const checkBoxSections = [];
+  const inputSections = [];
+
+  let currentFilterValue;
+  if (filterCategories) {
+    /* checkbox sections */
+    filterCategories.forEach((categoryConfig) => {
+      currentFilterValue = filterState[resourceType][categoryConfig.value];
+      if (categoryConfig.type === FilterType.MULTI_SELECT_VALUE) {
+        checkBoxSections.push({
+          title: categoryConfig.displayName,
+          categoryId: categoryConfig.value,
+          helpText: categoryConfig.helpText,
+          properties: categoryConfig.options.map((option) => {
+            return {
+              value: option.value,
+              labelText: option.displayName,
+              /* tertiary statement allows for 'false' to be explicity set as opposed to 'undefined' */
+              checked: currentFilterValue && currentFilterValue[option.value] ? true : false,
+            };
+          })
+        });
       }
-    ]
-  };
-};
+    });
 
-/*
-  TODO: Dispatch a real action
-*/
-export const mapDispatchToProps = (dispatch: any) => {
-  // return bindActionCreators({ onFilterChange } , dispatch);
+    /* input sections */
+    filterCategories.forEach((categoryConfig) => {
+      currentFilterValue = filterState[resourceType][categoryConfig.value] || '';
+      if (categoryConfig.type === FilterType.SINGLE_VALUE) {
+        inputSections.push({
+          categoryId: categoryConfig.value,
+          helpText: categoryConfig.helpText,
+          title: categoryConfig.displayName,
+          value: currentFilterValue,
```

Comment:
I will look into moving the appropriate pieces of logic into `CheckboxFilter`, `FilterSection`, and `InputFilter`, but I wouldn't want to intentionally tie the shape of the component props to what's in the AppConfig because either piece of data can change to add information. 

So even if the object key/value pairs up being the exact same after modifications, I would still explicitly map the properties instead of passing the entire object directly.

Reply:
For example I think it's confusing to have `categoryId: categoryConfig.value, value: currentFilterValue`.

```
{
  value: 'schema',
  displayName: 'Schema',
  helpText: 'help text',
  type: FilterType.SINGLE_VALUE,
},
```
I think calling it `categoryId` in the config is more descriptive. If we later decide to add a `defaultValue` for a filter that indicates which options are pre-checked, `value` doesn't align. Regarding `displayName` and `title`, I would prefer `label`.
```
{
  categoryId: 'schema',
  displayName: 'Schema',
  helpText: 'help text',
  type: FilterType.SINGLE_VALUE,
},
```

Reply:
@danwom I have modified the configuration to take `categoryId`. I agree that it should be the same name throughout tp avoid confusion, and I am also making the change in the redux layer as well where I just used `category`. 

I would actually prefer `displayName`. The reason is that we already have the existing precedent of using `displayName` in our configurations for cases where we don't want to use the code value as the UI value. I wouldn't want to introduce a new name for the same concept. If you believed all `displayName` values in the configurations should all be named `label` though, that's another story but changing it without a justification would inconvenience the community if they'd written any custom code with the previous configurations that use `displayName`

Reply:
I don't really have a preference of `label` vs `displayName` as long as we're consistent.